src/Controller/Api/FileManager.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+namespace App\Controller\Api;
+
+use App\Entity\DatasetSubmission;
+use App\Entity\File;
+use App\Entity\Fileset;
+use App\Util\Datastore;
+use Doctrine\ORM\EntityManagerInterface;
+use FOS\RestBundle\Controller\AbstractFOSRestController;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\IsGranted;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\BadRequestHttpException;
+use Symfony\Component\Routing\Annotation\Route;
+
+/**
+ * FileManager API used by File Manager app.
+ */
+class FileManager extends AbstractFOSRestController
+{
+    /**
+     * Delete a file or folder.
+     *
+     * @param DatasetSubmission      $datasetSubmission The id of the dataset submission.
+     * @param Request                $request           The request body
+     * @param EntityManagerInterface $entityManager     Entity manager interface instance.
+     * @param Datastore              $datastore         Datastore to manipulate the file on disk.
+     *
+     * @Route(
+     *     "/api/file_delete/{id}",
+     *     name="pelagos_api_file_delete",
+     *     methods={"DELETE"},
+     *     defaults={"_format"="json"},
+     *     requirements={"id"="\d+"}
+     *     )
+     *
+     * @IsGranted("IS_AUTHENTICATED_FULLY")
+     * @IsGranted("CAN_DELETE", subject="datasetSubmission")
```

Comment:
Should this voter subject be changed? I mean this actually works. We may not really need a file voter if the submission Id is passed along.